flake.nix

```diff
@@ -0,0 +1,253 @@
+{
+  description = "Emacs versions for CI";
+
```

Comment:
You could add the following config to this flake to enable the binary cache during the operation:

```nix
  nixConfig = {
    extra-substituters = ["https://emacs-ci.cachix.org"];
    extra-trusted-public-keys = ["emacs-ci.cachix.org-1:B5FVOrxhXXrOL0S+tQ7USrhjMT5iOPH+QN9q0NItom4="];
  };
```

Reply:
These are great suggestions, yay, thanks!